aiida/repository/repository.py

```diff
@@ -161,7 +162,7 @@ def _insert_file(self, path: pathlib.Path, key: str):
         if path.parent:
             directory = self.create_directory(path.parent)
         else:
-            directory = self.get_directory
+            directory = self.get_directory()
```

Comment:
this was a definite bug

Reply:
should we add the missing test perhaps then?

Reply:
ok I'll have a look 👍

Reply:
How was this intented to work? I think its actually impossible to reach the code, given that `Path().parent == Path()`. Is that perhaps what the test should be?

Reply:
or shall I just remove the if/else?

Reply:
I think the conditional makes sense it is just incorrect. What I think I intended here is to check whether the parent directory of the `path` exists. If not, then we create it first. Otherwise, we just get the directory.

That being said, "creating" directories here is all just virtual, and there aren't actual directories being explicitly created in the backend repository, just the virtual hierarchy. Since `create_directory` should only create the directory if it doesn't already exist, we could always just call it and don't need to check. So I guess we can indeed just get rid of the entire conditional and just call `self.create_directory(path.parent)`.

Reply:
yeh thats what I thought, removed